core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -304,13 +305,22 @@ class KafkaServer(val config: KafkaConfig, time: Time = Time.SYSTEM, threadNameP
           authZ
         }
 
+        auditor = try {
+          CoreUtils.createObject[Auditor](config.auditorClassName)
+        } catch {
+          case e: Exception =>
+            error("Creating auditor instance from the given class name failed.", e)
+            new NoOpAuditor
```

Comment:
Minor: maybe we can just make the auditor field optional in KafkaApis.scala so that we don't require to provide an `auditorClassName` config.

Reply:
Thanks for pointing it out. The alternative way (as you mentioned) is used to configure the authorizer. I did put some thoughts on whether to go with that or the current way to configure the auditor and I prefer the current way since I like making configuration properties explicit more than making them "optional".